### ISSUE TITLE
[Question] Is there a way to align `FontChoice` font ?

### ISSUE THREAD

**Issue by fengyichui** [mintty/wsltty#341]
Hi,

Some characters (I used font is `Iosevka`) are much higher than my CJK (font is `Microsoft YaHei UI`) characters, look like follow:
![font](https://github.com/mintty/wsltty/assets/17896291/47daf70c-8bb2-4d9c-9753-287531eb4dff)

My font config:
```
Font=Iosevka Term
FontHeight=12
FontChoice=CJK:5
Font5=Microsoft YaHei UI
```

Is there a way (or options) to pan up CJK font a bit or align those different font?

**Comment by mintty** [comment]
Please provide a download link to the Iosevka Term font. How does it look with other fonts?

**Comment by mintty** [comment]
Asking because it looks fine for me with most fonts, including Iosevka. It may be a font problem of Iosevka Term.

**Comment by fengyichui** [comment]
The download link is [here](https://github.com/be5invis/Iosevka/blob/v27.0.0/doc/PACKAGE-LIST.md). The `Iosevka Term` and `Iosevka` looks very similar and they are all more taller than my CJK font. The very old version of this font looks well.

**Comment by fengyichui** [comment]
I try Windows Font fallback:
```bat
 REG ADD "HKLM\Software\Microsoft\Windows NT\CurrentVersion\Fontlink\SystemLink" /v Iosevka /t REG_MULTI_SZ /d "MSYH.TTC,Microsoft YaHei UI,128,96"\0"MSYH.TTC,Microsoft YaHei UI"\0"SEGOEUI.TTF,Segoe UI,120,80"\0"SEGOEUI.TTF,Segoe UI"\0"SIMSUN.TTC,SimSun"\0"MSJH.TTC,Microsoft Jhenghei UI"\0"MEIRYO.TTC,Meiryo UI"\0"MALGUN.TTF,Malgun Gothic,128,96"\0"MALGUN.TTF,Malgun Gothic"\0"YUGOTHM.TTC,Yu Gothic UI,128,96"\0"YUGOTHM.TTC,Yu Gothic UI"\0"SEGUISYM.TTF,Segoe UI Symbol"
```

And remove wsltty config:
```
FontChoice=CJK:5
Font5=Microsoft YaHei UI
```

Those English and CJK font looks good (aligned):
![1](https://github.com/mintty/wsltty/assets/17896291/60405b7d-7b6b-451f-a860-2f16968c9c24)

**Comment by mintty** [comment]
I installed Iosevka Term from your link (ttf archive in case that matters) and used your config which looks good here:
![grafik](https://github.com/mintty/wsltty/assets/12740416/fca39cfb-6a05-4ce7-9c7e-ca413edb26f1)
Microsoft Yahei UI version is 6.30.

**Comment by fengyichui** [comment]
Did you uninstall all older versions of Iosevka before installing it?
My Microsoft Yahei UI version is 6.24, where do you download the version 6.30?

**Comment by mintty** [comment]
Iosevka Term was not installed before. Yahei UI came with Windows, maybe you should run an update.

**Comment by fengyichui** [comment]
I found this problem related to `FontHeight`, if it is equal to 12 it look bad, equal to 14 is no problem.
What is your configuration FontHeight?

**Comment by mintty** [comment]
I can reproduce the issue, kind of. But it depends on the actual characters used, how this looks:
![grafik](https://github.com/mintty/wsltty/assets/12740416/8c7168c9-524e-4329-8314-122211f9960f)
With this example, it seems that CJK characters also have some descender, like Latin. In this picture there is even more descender in the Latin g than in the CJK characters. Checking the baseline, there is 1 pixel offset. I guess 1 pixel is nothing that can be completely excluded in rendering, given rounding errors.
The CJK in this picture could be scaled a bit larger, though, but that's done by Windows.

**Comment by fengyichui** [comment]
Thanks! Using Windows Font Fallback is also a good choice.

**Comment by mintty** [comment]
For the records, DPI (display "scaling") also has an effect; my first screenshot (good) was at 125%, my second (confirming the issue) at 100% with 12pt font size.

**Comment by fengyichui** [comment]
This may be a problem with Windows font rendering, and Windows Font Fallback may have done some special processing internally.

**Comment by mintty** [comment]
The CJK font could be created a pinch bigger, like 1 pixel for 12pt font size, to compensate the problem.
However, it seems impossible to find heuristic criteria when to apply this tuning.
The same modification that helps for YaHei makes things worse for Yu Gothic and they have the same font metrics.

I'm considering a mechanism instead by which a specific font can be tuned a bit bigger by configuration.

**Comment by mintty** [comment]
Released 3.7.0.
